perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_eduPersonScopedAffiliations.java

```diff
@@ -20,4 +45,73 @@ public String getDestinationAttributeFriendlyName() {
 		return "eduPersonScopedAffiliations";
 	}
 
+	@Override
+	public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDefinition destinationAttributeDefinition) throws InternalErrorException {
+		//get already filled value obtained from UserExtSources
+		Attribute attribute = super.getAttributeValue(sess, user, destinationAttributeDefinition);
+
+		Attribute destinationAttribute = new Attribute(destinationAttributeDefinition);
+		//get values previously obtained and add them to Set representing final value
+		//for values use set because of avoiding duplicities
+		Set<String> valuesWithoutDuplicities = new HashSet<>();
+		if (attribute != null) {
+			Object value = attribute.getValue();
+			if (value != null && value instanceof Set) {
+				Object[] valArray = ((Set) value).toArray();
+				if (valArray instanceof String[]) {
+					for (String val: (String[]) valArray) {
+						valuesWithoutDuplicities.add(val);
+					}
+				}
+			}
+		}
+
+		Attribute attribute1 = null;
+		try {
+			//get value from urn:perun:user:attribute-def:def:eduPersonScopedAffiliationsManuallyAssigned
+			attribute1 = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, SOURCE_ATTRIBUTE_FRIENDLY_NAME_EPSA_MA_URN);
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("cannot read " + SOURCE_ATTRIBUTE_FRIENDLY_NAME_EPSA_MA + " of user " + user.getId(), e);
```

Comment:
This is not the right way how to catch and process WrongAttributeAssignment exception. Wrong assignment is probably error in the code itself so it can be thrown as InternalErrorException if needed.
For AttributeNotExists exception you need to choose if it is mandatory attribute or optional attribute for the purpose of getting final value.
If it is mandatory, then you can throw ConsistencyErrorException instead.
If it is optional, you can use more precise message like "Attribute .... not exists so it's values will be skipped" and log it as debug, not as error.